liblwgeom/lwrandom.c

```diff
@@ -0,0 +1,278 @@
+/**********************************************************************
+ *
+ * PostGIS - Spatial Types for PostgreSQL
+ * http://postgis.net
+ *
+ * PostGIS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * PostGIS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PostGIS.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **********************************************************************
+ *
+ * Copyright 2019 Mike Taves
+ *
+ **********************************************************************/
+
+
+#include <stdlib.h>
+#include <stdint.h>
+/* #include <stdio.h> */
+
+#include "lwrandom.h"
+#include <stdlib.h>
+#include <time.h>
+
+
+static unsigned char _lwrandom_seed_set = 0;
+static int32_t _lwrandom_seed[3] = {0x330e, 0xabcd, 0x1234};
+
+/*
+ * Set seed for a random number generator.
+ * A zero value uses clock as seed the first time only.
+ */
+void
+lwrandom_set_seed(uint32_t seed)
```

Comment:
Is there any reason to use `uint32_t` instead of `int32_t`. It's stored as `int32_t` and read as a `int32_t` but casted implicitly in https://github.com/postgis/postgis/pull/365/files#diff-978bce4984e620a04ff984c6c12d0776R1037.

Reply:
No big reason, but perhaps I'll change these to `int32_t` for consistency. I know that ` _lwrandom_seed[3]` (and similar) cannot be `uint32_t`, as I was getting unexpected results.